lib/msal-browser/src/config/Configuration.ts

```diff
@@ -310,11 +325,25 @@ export function buildConfiguration(
         },
     };
 
+    // Throw an error if user has set OIDCOptions without being in OIDC protocol mode
+    if(userInputAuth?.protocolMode !== ProtocolMode.OIDC && 
+        userInputAuth?.OIDCOptions) {
+            throw ClientConfigurationError.createCannotSetOIDCOptionsError();
+    }
+
+    // Throw an error if user has set allowNativeBroker to true without being in AAD protocol mode
+    if(userInputAuth?.protocolMode &&
+        userInputAuth.protocolMode !== ProtocolMode.AAD &&
+        providedSystemOptions?.allowNativeBroker) {
+            throw ClientConfigurationError.createCannotAllowNativeBrokerError();
+    }
+
     const overlayedConfig: BrowserConfiguration = {
-        auth: { ...DEFAULT_AUTH_OPTIONS, ...userInputAuth },
+        auth: { ...DEFAULT_AUTH_OPTIONS, ...userInputAuth as InternalAuthOptions },
```

Comment:
We should not be typecasting the user input, if this is needed to pass typechecking something is not quite right above
```suggestion
        auth: { ...DEFAULT_AUTH_OPTIONS, ...userInputAuth },
```

Reply:
I had it like this, because otherwise, Typescript would throw an error that the types are not assignable to each other.

Reply:
The problem in this case is that you start with the default (which has everything defined) then you're overwriting the defaults with whatever the user provided. If, let's say, the user doesn't provide default scopes your final object will have default scopes undefined because the whole default object gets overwritten with the object the user provided.

You could address this by doing something like this:

```js
const { OIDCOptions as userInputOIDCOptions, ...restUserInput } = userInputAuth;

...
auth: {
 ...DEFAULT_AUTH_OPTIONS, 
 ...restUserInput, 
 OIDCOptions: { ...DEFAULT_AUTH_OPTIONS.OIDCOptions, ...userInputOIDCOptions }
}
```

You don't see this pattern in the existing logic because we haven't really done nested objects like this before, instead typically opting to create a new category if config options make sense to be grouped together like this. That said I'm ok with either, I'll leave it up to you what you prefer.

Reply:
The stuff about the nested objects makes sense. I'd prefer to have OIDCOptions in auth (to me, it makes the most sense to set ProtocolMode and OIDCOptions in the same object), so I implemented something like this. Thank you!